### PR TITLE
fix: context fallback to user instead of undefined

### PR DESCRIPTION
**Pull request by kulmann**
## Description
We've fixed a bug where routes without explicit `auth` requirement (i.e. user context) and without any context route in the URL were recognized as neither user-context nor public-link-context. In such situations we now expect that the session requires a user and redirect to the login page (instead of staying stuck on a most likely broken page).

## Motivation / context
We want to provide an endpoint in `ocis` that let's e.g. the desktop client query a URL for opening a document (via app provider / external app) in web. The desktop client has the file id and the app provider is capable of determining a default app if no app name is provided. But neither the desktop client nor the ocis backend know what route the file might have come from if opened via files app. Opening a file in web from the desktop client is always supposed to open in an authenticated session / user session (user needs to log in if not already done before).

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Review by dschmidt — APPROVED** [review]

**Comment by dschmidt** [comment]
Interesting unit test failure in `FileDetails.spec.js`:

Screenshot_20220812_091101


 That happens because

```
    showShares() {
      return this.hasAnyShares && isUserContext(this.$router, this.$route)
    },
``` 

now returns true in this test. I think it's fine to just update the snapshots... 

But this breaks the AppTopBar! We need to adjust the TopBar to not show/disable the close button (and explain that) or make it work somehow for example by defaulting the close to navigating to `/`. 
I would ask you to take care of that in this PR so we don't forget about it, as it shouldn't be a huge change anyway.

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/27511/12/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27511/tracing/participant-management-alice-2022-8-12-07-36-49.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27511/tracing/participant-management-anonymous-2022-8-12-07-37-03.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27511/tracing/participant-management-brian-2022-8-12-07-36-53.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27511/tracing/participant-management-carol-2022-8-12-07-36-58.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27511/tracing/public-link-alice-2022-8-12-07-34-53.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27511/tracing/public-link-anonymous-2022-8-12-07-35-05.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27511/tracing/unstructured-collection-of-testable-space-interactions-alice-2022-8-12-07-38-40.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27511/tracing/unstructured-collection-of-testable-space-interactions-anonymous-2022-8-12-07-39-18.zip 


</p></details>

**Comment by kulmann** [comment]
> isUserContext

I've changed the implementation to not use the `isUserContext` helper anymore (that's meant for internal auth handling only) but instead use the `usePublicLinkContext` composable.

Regarding the close action I've added a default of redirecting to `/` if the current file context has no context route name set. Good finding!

**Comment by kulmann** [comment]
Some issues remain with shares loading in the file details panel. At least it looks weird to me. Will have a look tomorrow morning.

**Comment by dschmidt** [comment]
> > isUserContext
> 
> I've changed the implementation to not use the `isUserContext` helper anymore (that's meant for internal auth handling only) but instead use the `usePublicLinkContext` composable.

👍🏻 

> Regarding the close action I've added a default of redirecting to `/` if the current file context has no context route name set. Good finding!

Hehe, yeah, I care about apps 🤓 Good fix!

File: packages/web-app-files/src/components/SideBar/Details/FileDetails.vue [hunk 1]
```diff
@@ -430,13 +430,20 @@ export default defineComponent({
...
     async loadData() {
+      if (this.file.type === 'folder' || this.isPublicLinkContext) {
+        return
+      }
       this.loading = true
-      const calls = []
-      if (this.file.type === 'file' && isUserContext(this.$router, this.$route)) {
-        calls.push(this.loadVersions({ client: this.$client, fileId: this.file.id }))
+      try {
+        await Promise.allSettled([
```
Comment (dschmidt) [inline packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:438]: Why allSettled? Can the catch block be entered at all then?

Comment (kulmann) [reply packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:438]: hm true, probably not...

Comment (kulmann) [reply packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:438]: Running into a major scope creep... reverting that change and following up in a separate PR.

Comment (dschmidt) [reply packages/web-app-files/src/components/SideBar/Details/FileDetails.vue]: 👍🏻

**Pull request merged**